### ISSUE TITLE
Task not found in Rails 7 app

### ISSUE THREAD

**Issue by mdesantis** [railsware/js-routes#294]
I'm trying to use js-routes in a Rails 7 app, but `bundle exec rake js:routes` task isn't found:

```
> bundle exec rake js:routes
rake aborted!
Don't know how to build task 'js:routes' (See the list of available tasks with `rake --tasks`)
ruby/3.0.3/bin/bundle:23:in `load'
ruby/3.0.3/bin/bundle:23:in `<main>'
```

 It works correctly if the task file is copied from the library to `lib/tasks` application directory.

**Comment by mdesantis** [comment]
> Did you manage to find a solution for this ?

No, I just copied the task file from the library to `lib/tasks` application directory

**Comment by Marcellllll** [comment]
Looks like it is something in Rails 7 which is breaking things.  With Rails 6 sprockets can find my js-routes, with rails 7 it can no longer find it.

**Comment by wtfspm** [comment]
In my app (Rails 7, no Sprockets), it was enough to move the Sprockets conditional from `lib/js_routes.rb` to `lib/js_routes/engine.rb` so the engine is still defined and Rails can pick up the tasks.

```
# lib/js_routes.rb

if defined?(::Rails)                 ## move this: defined?(::Sprockets::Railtie)
  require 'js_routes/engine'
end
```

```
# lib/js_routes/engine.rb

class Engine < ::Rails::Engine
  if defined?(::Sprockets::Railtie)  ## to here
    require 'sprockets/version'
    ...
```

**Comment by wtfspm** [comment]
Yeah, it's hard to know what everyone's setup is. When you said you generated a new Rails 7 app, I assumed you included Sprockets since it's the default, and with Sprockets defined, the engine is loaded, so no issue. Happy to submit a PR, though I'm curious if this fixes the issue for @mdesantis or @Marcellllll

**Comment by bogdan** [comment]
> When you said you generated a new Rails 7 app, I assumed you included Sprockets since it's the default, and with Sprockets defined, the engine is loaded, so no issue

Hm, you are right. It was a surprise for me that `sprockets-rails` is included in `Gemfile` even in Rails 7.x.

Merged your PR and released js-routes `2.2.3`.
